internal/v1/handler_blueprints.go

```diff
@@ -17,6 +20,10 @@ import (
 	"github.com/labstack/echo/v4"
 )
 
+const (
+	blueprintNameRegex = `^[a-zA-Z0-9][a-zA-Z0-9-_ ]+[a-zA-Z0-9]$`
```

Comment:
I suggest to always put dash as the last character because this can accidentally create range.

Is this because this is turned into compose name which has these constraints?

Reply:
nope, just me adding the regexp manually randomly ... happy to switch around
I've excluded special characters here just for convenience, but I'm transposing to image name here: #1093 so we can just drop the special characters there and allow anything here as long as it starts with word? :shrug:

Reply:
While thinking about it, I think we can allow smiley faces in the name, might be cool and doesn't hurt anything.
Let's go with just requiring non-empty (contains word)

Reply:
If this is not coming from an existing constraint, everything in 2024 is UTF8 and I would not filter out non-ASCII languages like Japanese or Chinese for no reason.

Reply:
in #1093 I'm transposing it to machine friendly (asci) image name, so if there is no ascii chars, it will result in `___-____-___-____` image name :thinking: not sure if that's what we want xD
But why not :)